editor/project_settings_editor.cpp

```diff
@@ -809,28 +811,41 @@ void ProjectSettingsEditor::_item_del() {
 	undo_redo->commit_action();
 }
 
-void ProjectSettingsEditor::_action_adds(String) {
+void ProjectSettingsEditor::_action_check(String p_action) {
 
-	_action_add();
-}
+	if (p_action == "") {
 
-void ProjectSettingsEditor::_action_add() {
+		action_add->set_disabled(true);
+	} else {
 
-	String action = action_name->get_text();
-	if (action.find("/") != -1 || action.find(":") != -1 || action == "") {
-		message->set_text(TTR("Invalid action (anything goes but '/' or ':')."));
-		message->popup_centered(Size2(300, 100) * EDSCALE);
-		return;
+		if (p_action.find("/") != -1 || p_action.find(":") != -1) {
+			action_add->set_text(TTR("Can't contain '/' or ':'."));
+			action_add->set_disabled(true);
+			return;
+		}
+		if (ProjectSettings::get_singleton()->has("input/" + p_action)) {
+			action_add->set_text(TTR("Added"));
```

Comment:
"Already existing" would be clearer no?

Reply:
I'm not sure. For me, if I type in an action name, press enter and it says "Already existing", I'd think I've added it before.

Reply:
Well that's what it means no? :)

Reply:
Once you've added an action, `action_add` should be cleared, so you don't need to handle this case. Then if you test for the action already existing, that's what the error should say.